astropy/coordinates/representation.py

```diff
@@ -478,12 +612,74 @@ def represent_as(self, other_class):
         ----------
         other_class : `~astropy.coordinates.BaseRepresentation` subclass
             The type of representation to turn the coordinates into.
+        differential_class : dict, `~astropy.coordinates.BaseDifferential` (optional)
+            TODO
         """
-        if other_class is self.__class__:
-            return self
+        if other_class is self.__class__ and not differential_class:
+            return self.without_differentials()
+
         else:
+            if isinstance(other_class, six.string_types):
+                raise ValueError("Input to a representation's represent_as "
+                                 "must be a class, not a string. For "
+                                 "strings, use frame objects")
+
             # The default is to convert via cartesian coordinates
-            return other_class.from_cartesian(self.to_cartesian())
+            new_rep = other_class.from_cartesian(self.to_cartesian())
+            new_rep._differentials = self._re_represent_differentials(
+                new_rep, differential_class)
+
+            return new_rep
+
+    def with_differentials(self, differentials):
+        """
+        Create a new representation with the same positions as this
+        representation, but with these new differentials.
+
+        Differential keys that already exist in this object's differential dict
+        are overwritten.
+
+        Parameters
+        ----------
+        differentials : Sequence of `~astropy.coordinates.BaseDifferential`
+            The differentials for the new representation to have.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but with the ``differentials`` as
+            its differentials.
+        """
+        if not differentials:
+            return self
+
+        args = [getattr(self, component) for component in self.components]
+
+        # We shallow copy the differentials dictionary so we don't update the
+        # current object's dictionary when adding new keys
+        new_rep = self.__class__(*args, differentials=self.differentials.copy(),
+                                 copy=False)
+        new_rep._differentials.update(
+            self._validate_differentials(differentials))
+
+        return new_rep
+
+    def without_differentials(self):
+        """
+        Create a new representation with the same positions as this
+        representation, but without any differentials.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but without the ``differentials``.
```

Comment:
Maybe explicitly say "shallow copy"? In any case, definitely state that one gets the original back if it has no differentials attached (which I think is correct behaviour).